wrapper/qsl/src/main/kotlin/org/quiltmc/qsk/wrapper/qsl/recipe/RecipeExtensions.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsk.wrapper.qsl.recipe
+
+import net.minecraft.inventory.Inventory
+import net.minecraft.item.Item
+import net.minecraft.item.ItemConvertible
+import net.minecraft.item.ItemStack
+import net.minecraft.recipe.*
+import net.minecraft.tag.TagKey
+import net.minecraft.util.Identifier
+import net.minecraft.util.registry.Registry
+import org.quiltmc.qsl.recipe.api.RecipeManagerHelper
+import org.quiltmc.qsl.recipe.api.builder.ShapedRecipeBuilder
+import org.quiltmc.qsl.recipe.api.builder.ShapelessRecipeBuilder
+import org.quiltmc.qsl.recipe.api.builder.VanillaRecipeBuilders.*
+
+/**
+ * Currently Kotlin does not have union types, so this is a workaround.
```

Comment:
this could do with some extra documentation detailing *which* types are supported as ingredients (a `Ingredient`, an array of `ItemConvertible`s or `ItemStack`s, etc.)

though it would be cool to express them like
```kt
public typealias IngredientItemLike = ItemConvertible | ItemStack
public typealias IngredientLike = Ingredient | Array<IngredientItemLike> | Collection<IngredientItemLike> | TagKey<*>
```